packages/@aws-cdk/aws-codestarnotifications/lib/rule.ts

```diff
@@ -0,0 +1,262 @@
+import { Stack, IResource, Resource, Names } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnNotificationRule } from './codestarnotifications.generated';
+import { RuleSourceConfig, SourceType, IRuleSource } from './source';
+import { IRuleTarget, RuleTargetConfig } from './target';
+
+/**
+ * The level of detail to include in the notifications for this resource.
+ */
+export enum DetailType {
+  /**
+   * BASIC will include only the contents of the event as it would appear in AWS CloudWatch
+   */
+  BASIC = 'BASIC',
+
+  /**
+   * FULL will include any supplemental information provided by AWS CodeStar Notifications and/or the service for the resource for which the notification is created.
+   */
+  FULL = 'FULL',
+}
+
+/**
+ * The status of the notification rule.
+ */
+export enum Status {
+
+  /**
+   * If the status is set to DISABLED, notifications aren't sent.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * If the status is set to ENABLED, notifications are sent.
+   */
+  ENABLED = 'ENABLED',
+}
+
+/**
+ * The options for AWS Codebuild and AWS Codepipeline notification integration
+ */
+export interface RuleProps {
```

Comment:
This is the same as `NotifyOnEventOptions`, right? You should make this one extend `NotifyOnEventOptions`.

Reply:
Already adjusted.